components/WeatherForecast.js

```diff
@@ -0,0 +1,118 @@
+import {
+  WeatherForecastField,
+  Overlay,
+  StyledDiv,
+  DisplayedDayDetailsPage,
+  ThreeDaysContainer,
+  WeatherIconDetailsPage,
+  DayContainer,
+  ModalContainer,
+  TemperatureDetailsPage,
+  WeatherHeadline,
+} from "./styledComponents/WeatherForecast.styles";
+
+import { useState, useEffect } from "react";
+import useSWR from "swr";
+import WeekdayFromDateString from "./WeekdayFromDateString";
+import WeatherForecastModal from "./WeatherForecastModal";
+
+const API = process.env.NEXT_PUBLIC_WEATHER_API_KEY;
```

Comment:
Here you are sending your API Key to the client (it is in a component that is rendered into the app). This is something we want to prevent by any means, since this is very sensitive information which we want to keep private. We can do that by not sending it to the client (aka not using it in any component that is rendered to our App). 

The solution here is to create an API endpoint, which are NOT send to the client but are run on our own server. You can send the lat and long information to your own endpoint like this: 

```js
const { data: weather } = useSWR('/api/weather/?lat=${lat}&long=${long}`)
```
(please note that the variables lat and long need to be exchanged with your actual latitude and longitude data)

In your API endpoint, you can then create the fetch request to the openweathermap API including the sensitive Key:

```js
// api/weather.js
...
if(request.method === 'GET') {
  const apiResponse =  await fetch(
    `https://api.openweathermap.org/data/2.5/forecast? 
    lat=${detailActivity.lat}&lon=${detailActivity.lng}&units=metric&appid=${API}`
  )
  const data = await apiResponse.json()
  // send the data back to your client.
}
```
Hope this helps